### PR TITLE
Hud icons are now based on ID card styles (+crew monitor, crew manifest are based on hud icons than job names + chameleon agent card QoL)

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
tracking Hud icon based on ID card style

## Why It's Good For The Game
a basic start to implementing alternative job title system.
Also, you might want to give a crew a special job, but seeing them as a grey unknown hud icon is kinda annoying.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://youtu.be/qMRL6g63jWA

(ignore some weird manipulations. They're fixed.)

image

Job title is Captain, but HUD is security officer because it's Sec card style.

image

crew monitor tracks based on hud icon now

image
image

crew manifest is based on hud icon department type too (regardless of job name)

-------------

image

chameleon QOL
Once you change a card into a job card, you'll get a pop up, and if you click yes, you get the default job title of the chosen card.
if there's no default name or you chose no, you'll still get the previous job title.

--------
https://www.youtube.com/watch?v=b6RU4wTuUv0&feature=youtu.be

broken chameleon agent card

</details>

## Changelog
:cl: EvilDragonfiend, PestoVerde322
add: Hud is now tracked based on your card appearance. HoP is now supposed to repaint crews' card in order to give them correct hud icons, or securities(+ghosts) will blame you.
add: every card now has default hud icons. (i.e. Captain's Spare card has a blank command HUD, budget cards have blank department color HUD)
add: added more PDA style options to PDA painter. (deputy, prisoner, heads, unassigned, etc)
refactor: PDA painter is now easy to recognise colourising style easier.
tweak: Some job HUDs are now only available through emaged ID card repainter.
tweak: crew monitor now tracks crew job color based on hud state.
tweak: Assistant id card crew objective is now greentextable when your card is painted even if the job name is still assistant.
add: hud image based defines
refactor: crew manifest is based on hud state.
refactor: PDA painter machine now updates the crew manifest upon ID repainting
refactor: changes to chameleon code and agent card, to make Hud icon update working. Chameleon agent card has earned QoL. They'll get a default job name of a card type you chose once changed.
add: agent card will be randomised upon EMP. (random hud icon, random job name, random icon)
add: broken chameleon agent card (with all wiped access, and unable to copy accesses from another card)
add: broken chameleon kit will have broken chameleon agent card.
add: new hud icons, new card images (blank symbol department color hud icons, new hudking icon)
del: 9x9 pixel size 'hudking' hud icon
add: New card types for the purpose of chameleon agent card. (i.e. "their majesty's gold identification card")
tweak: Some card types are now blacklisted from chameleon agent card - because they don't make sense. (i.e changeling fake ones)
tweak: your card paycheck department is now re-assigned based on your hud icon state. if your hud icon is red with job name 'not real security', you are paid by the security budget. the process happens when you repaint a card or give a new job name to a card.
tweak: not all gimmicks are in civilian department anymore. Psychiatrist is medical, and Stage Magician is service like their mime/clown fellows. (VIP and barber are still service)
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/crew.dm [hunk 1]
```diff
@@ -190,10 +198,10 @@ GLOBAL_DATUM_INIT(crewmonitor, /datum/crewmonitor, new)
 		var/obj/item/card/id/I = tracked_human.wear_id ? tracked_human.wear_id.GetID() : null
 
 		if (I)
-			entry["name"] = I.registered_name
-			entry["assignment"] = I.assignment
-			if(jobs[I.assignment] != null)
-				entry["ijob"] = jobs[I.assignment]
+			entry["name"] = I.registered_name ? I.registered_name : "Unknown"
+			entry["assignment"] = I.assignment ? I.assignment : "Unknown"
```
Comment (EvilDragonfiend) [inline code/game/machinery/computer/crew.dm:203]: card with no name/assignment makes bad null bug here.

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -345,45 +357,19 @@ update_label("John Doe", "Clowny")
...
 	access = list(ACCESS_PIRATES)
+	hud_state = "syndicate"
 
 /obj/item/card/id/syndicate
 	name = "agent card"
 	access = list(ACCESS_MAINT_TUNNELS, ACCESS_SYNDICATE)
+	hud_state = "syndicate"
 	var/anyone = FALSE //Can anyone forge the ID or just syndicate?
 	var/forged = FALSE //have we set a custom name and job assignment, or will we use what we're given when we chameleon change?
-	var/static/list/available_icon_states = list(
```
Comment (EvilDragonfiend) [inline code/game/objects/items/cards_ids.dm:358]: actually not used variable

**Comment by Crossedfall** [comment]
Good change

File: code/game/machinery/PDApainter.dm [hunk 3]
```diff
@@ -43,38 +43,25 @@
...
-		"Geneticist" = "gene",
-		"Brig Physician" = "brigphys",
-		"Deputy" = "deputy",
-		"Roboticist" = "roboticist",
-		"Janitor" = "janitor",
-		"Exploration Crew" = "exploration"
+
+	var/valid_jobs = list(
+		// Command
+		"----Command----","Command (Custom)", "Captain", "Acting Captain",
```
Comment (Penwin0) [inline code/game/machinery/PDApainter.dm:57]: Won't this show "----Command----" on the list of options in the PDA painter? What happens if I try to pick it?

File: code/game/machinery/PDApainter.dm [hunk 4]
```diff
@@ -267,3 +261,155 @@
 /obj/machinery/pdapainter/power_change()
 	..()
 	update_icon()
+
+
+/proc/get_cardstyle_by_jobname(jobname)
```
Comment (Penwin0) [inline code/game/machinery/PDApainter.dm]: I feel like both `get_cardstyle_by_jobname` and `get_hud_by_jobname` could both be assimilated into one, there's a lot of repetition, or just handled in `var/valid_jobs` like it was previously

File: code/game/machinery/PDApainter.dm (hunk 3 above)
Comment (EvilDragonfiend) [reply code/game/machinery/PDApainter.dm:57]: Does nothing. It's visual boundaries for the selection window.

File: code/game/machinery/PDApainter.dm (hunk 4 above)
Comment (EvilDragonfiend) [reply code/game/machinery/PDApainter.dm]: likely no.
```
(card style)
			"Botanist" = "serv",
			"Bartender" = "serv",
			"Cook" = "serv",
```
```
(hud style)
			"Bartender" = "bartender",
			"Cook" = "cook",
			"Botanist" = "botanist",
```
each of them has different values and they can't be put together. syncing their names isn't a good idea, and putting them into a single proc will make this more uglier as well, 
I don't like this too, but I don't have any particular idea to make it better pretty.

Comment (EvilDragonfiend) [reply code/game/machinery/PDApainter.dm]: I forgot to put another reason too.
`get_hud_by_jobname`
`get_cardstyle_by_jobname`
proc names are obvious to realise what they do.

**Comment by ItsTisme** [comment]
if your job is manualy changed it should prop update the card/icon(assistant being promoted to medical or jannitor as an example

**Comment by EvilDragonfiend** [comment]
> if your job is manualy changed it should prop update the card/icon(assistant being promoted to medical or jannitor as an example

I thought about it, but I want HoP to be sure about the importance of the painting machine, and make the field promotion less usual.

**Comment by EvilDragonfiend** [comment]
currently touching crew mamifest too

**Comment by EvilDragonfiend** [comment]
new change. now crew manifest is based on hud

<details>
<summary>Screenshots&Videos</summary>

image
image

crew manifest is based on hud icon department type too (regardless of job name)

</details>


Now this PR might be ready to take a full review now.

**Comment by EvilDragonfiend** [comment]
there will be no additional change from now. might be ready to take a review.

**Review by PowerfulBacon — COMMENTED** [review]
What happens if I wear a budget card or other card that isn't an ID?

**Comment by EvilDragonfiend** [comment]
> What happens if I wear a budget card or other card that isn't an ID?

You can't wear a card that their path isn't `/card/id/`.
budget card's path is `/card/id/` so, you can wear them, but basically `HUD_UNKNOWN` is assigned at default, while budget card has a blank department colour. For example, seccie card is just red.

**Comment by PowerfulBacon** [comment]
Seems good, their majesties card icon is too big though
image

**Comment by EvilDragonfiend** [comment]
> Seems good, their majesties card icon is too big though image

image
image

**Comment by EvilDragonfiend** [comment]
image

add: added more PDA style options to PDA painter. (deputy, prisoner, heads, unassigned, etc)


`PDA` pull is PDA painter QoL one, and is tested too. Not that different from the last testmerge.

**Comment by EvilDragonfiend** [comment]
due to the system how your card's department is assigned, more code edits happened. ahhhhhhhh----
two pulls were made from the last edit, and they're separately made for better maintainer's review if they took a look on mine already.

**Comment by St0rmC4st3r** [comment]
>Some job HUDs are now only available through emaged ID card repainter.

IMO this should be a hackable function, not emagged. The emag already has too much stuff tied to it.

**Comment by EvilDragonfiend** [comment]
> > Some job HUDs are now only available through emaged ID card repainter.
> 
> IMO this should be a hackable function, not emagged. The emag already has too much stuff tied to it.

Unless we have a better hacking design for overall, I think i would go this. We can change it later.

**Review by PowerfulBacon — APPROVED** [review]
Yolo, seems alright

**Pull request merged**